Update test_create_dataset_in_subfolder to check for the dataset presence

davelopez opened the pull request:
To avoid flakiness due to slight time differences on slow systems like CI.

```
FAILED lib/galaxy_test/api/test_libraries.py::TestLibrariesApi::test_create_dataset_in_subfolder - AssertionError: <Response [200]>
assert '2025-02-19T12:18:15.550553' == '2025-02-19T12:18:15.522299'
  - 2025-02-19T12:18:15.522299
  ?                      ^^^^^
  + 2025-02-19T12:18:15.550553
  ?                      ^^^^^
```

## How to test the changes?
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

davelopez committed:
Relax time comparison in test_create_dataset_in_subfolder

To avoid slight time differences on slow systems like CI.

nsoranzo commented on lib/galaxy_test/api/test_libraries.py:
```diff
@@ -427,10 +428,13 @@ def test_create_dataset_in_subfolder(self):
+            container_fetch_response.json()["folder_contents"][0]["update_time"]
+        )
+        # Assert that the update time of the dataset is approximately the same as the update time of the container
+        assert abs((dataset_update_time - container_update_time).total_seconds()) < 1
```
Wouldn't be better to just check that the subfolder contains the dataset, instead of looking at update times?

davelopez replied:
Yep, that would make more sense, but since the test is 6 years old I didn't know if there was a good reason for checking this so... :sweat_smile: 

I will drop this check and just assert the dataset is there :+1:

davelopez committed:
Check if the dataset is in the library folder

This check is more robust than the previous one,
which was based on the update time of the folder.

nsoranzo reviewed (approved):
Thanks @davelopez !
